independent-projects/qute/core/src/main/java/io/quarkus/qute/CheckedTemplate.java

```diff
@@ -113,4 +113,34 @@
      */
     String defaultName() default ELEMENT_NAME;
 
+    /**
+     * By default, a <code>native static</code> method with the name that contains a dollar sign {@code $} denotes a method that
+     * represents a fragment of a type-safe template. It's possible to ignore the fragments and effectively disable this
+     * feature.
+     * <p>
+     * The name of the fragment is derived from the annotated method name. The part before the last occurence of a dollar sign
+     * {@code $} is the method name of the related type-safe template. The part after the last occurence of a dollar sign is the
+     * fragment identifier - the strategy defined by the relevant {@link CheckedTemplate#defaultName()} is used.
+     * <p>
+     * Parameters of the annotated method are validated. The required names and types are derived from the relevant fragment
+     * template.
+     *
+     * <pre>
+     * &#64;CheckedTemplate
+     * class Templates {
+     *
+     *     // defines a type-safe template
+     *     static native TemplateInstance items(List&#60;Item&#62; items);
+     *
+     *     // defines a fragment of Templates#items() with identifier "item"
+     *     &#64;CheckedFragment
+     *     static native TemplateInstance items$item(Item item);
+     * }
+     * </pre>
```

Comment:
This appears to document type-safe fragments, which is good, but it documents it in the `ignoreFragments` method, which turns it off, so this is probably misplaced. The documentation for how to declare type-safe fragments should be on the main annotation class IMO.